test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,528 @@
+        let authority = if let Ok(authority) = parse_uauthority(&value) {
+            authority
+        } else {
+            let err_msg = "Error parsing authority: ".to_string();
+            error!("{}", err_msg);
+            UAuthority::default()
+        };
+
+        let resource = parse_uresource(&value);
+        let entity = parse_uentity(&value);
+
+        let uuri = if authority.get_name().is_none() && authority.number.is_none() {
+            UUri {
+                entity: MessageField(Some(Box::new(entity))),
+                resource: MessageField(Some(Box::new(resource))),
+                ..Default::default() // If authority is default, fill in the rest with default values
+            }
+        } else {
+            debug!(" authority is not default");
+            UUri {
+                authority: MessageField(Some(Box::new(authority))),
+                entity: MessageField(Some(Box::new(entity))),
+                resource: MessageField(Some(Box::new(resource))),
+                ..Default::default()
+            }
+        };
+
+        Ok(WrapperUUri(uuri))
+    }
+}
+
+fn parse_uresource(value: &Value) -> UResource {
+    let mut uresource = UResource::new();
+    if let Some(resource_value) = value
+        .get("resource")
+        .and_then(|resource_value| resource_value.get("name"))
+        .and_then(|resource_value| resource_value.as_str())
+    {
+        uresource.name = resource_value.to_owned();
+    } else {
+        error!("Error: name field is not a string in resource");
+    };
+
+    if let Some(resource_value) = value
+        .get("resource")
+        .and_then(|resource_value| resource_value.get("instance"))
+        .and_then(|resource_value| resource_value.as_str())
+    {
+        uresource.instance = Some(resource_value.to_owned());
+    } else {
+        error!("Error: instance field is not a string in resource");
+        //Some("None".to_owned())
+    };
+
+    if let Some(resource_value) = value
+        .get("resource")
+        .and_then(|resource_value| resource_value.get("message"))
+        .and_then(|resource_value| resource_value.as_str())
+    {
+        uresource.message = Some(resource_value.to_owned());
+    } else {
+        error!("Error: message field is not a string in resource_value");
+        //Some("None".to_owned())
+    };
+
+    if let Some(resource_value) = value
+        .get("resource")
+        .and_then(|resource_value| resource_value.get("id"))
+        .and_then(|resource_value| resource_value.as_str())
+    {
+        if let Ok(parsed_id) = resource_value.parse::<u32>() {
+            uresource.id = Some(parsed_id);
+        } else {
+            error!("Error: id field parsing to u32");
+            // Some(0)
+        }
+    } else {
+        error!("Error: id field is not string");
+        //Some(0)
+    };
+    uresource
+}
+
+fn parse_string_field(value: &Value, field: &str) -> Result<String, serde_json::Error> {
```

Comment:
Function name seems fairly generic, but the internals explicitly reference `entity_value`. Should probably change one or the other.